controllers/mirroring/mirroring_controller.go

```diff
@@ -0,0 +1,478 @@
+							cephBlockPool.Spec.Mirroring.Peers.SecretNames, secretName)
+					}
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf(
+				"failed to update CephBlockPool %s for mirroring: %w",
+				cephBlockPool.Name,
+				err,
+			)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *MirroringReconciler) reconcileRadosNamespaceMirroring(
+	ocsClient *providerClient.OCSProviderClient,
+	clientMappingConfig *corev1.ConfigMap,
+	storageClusterPeer *ocsv1.StorageClusterPeer,
+	disableMirroring bool,
+) (ctrl.Result, error) {
+	/*
+		Algorithm:
+		make a list of peerClientIDs
+		send GetStorageClientsInfo with this Info
+		make a map of remoteNamespaceByClientID from response
+		list all radosNamespace and for each,
+			find out which consumer does it belong to
+			find the localClientID and use map to get radosnamespce
+			if radosnamespace is empty, disable mirroring
+			else enable mirroring
+	*/
+
+	peerClientIDs := []string{}
+	storageConsumerByName := map[string]*ocsv1alpha1.StorageConsumer{}
+	for localClientID, peerClientID := range clientMappingConfig.Data {
+		// for internal mode, we need only blockPool mirroring, hence skipping this for the special key "internal"
+		if localClientID == internalKey {
+			continue
+		}
+		// Check if the storageConsumer with the ClientID exists, this is a fancy get operation as
+		// there will be only one consumer with the clientID
+		storageConsumers := &ocsv1alpha1.StorageConsumerList{}
+		if err := r.list(storageConsumers, client.MatchingFields{clientIDIndexName: localClientID}); err != nil {
+			return ctrl.Result{}, err
+		}
+		if len(storageConsumers.Items) != 1 {
+			return ctrl.Result{}, fmt.Errorf("expected 1 StorageConsumer but got %v", len(storageConsumers.Items))
+		}
+		storageConsumerByName[storageConsumers.Items[0].Name] = &storageConsumers.Items[0]
+		peerClientIDs = append(peerClientIDs, peerClientID)
+	}
+
+	response, err := ocsClient.GetStorageClientsInfo(
+		r.ctx,
+		storageClusterPeer.Status.PeerInfo.StorageClusterUid,
+		peerClientIDs,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	remoteNamespaceByClientID := map[string]string{}
+	for i := range response.ClientsInfo {
+		clientInfo := response.ClientsInfo[i]
+		remoteNamespaceByClientID[clientInfo.ClientID] = clientInfo.RadosNamespace
+	}
+
+	radosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+	if err = r.list(radosNamespaceList, client.InNamespace(storageClusterPeer.Namespace)); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to list CephBlockPoolRadosNamespace(s): %w", err)
+	}
+
+	for i := range radosNamespaceList.Items {
+		rns := &radosNamespaceList.Items[i]
+		consumer := storageConsumerByName[rns.GetLabels()[controllers.StorageConsumerNameLabel]]
+		if consumer == nil {
+			continue
+		}
+		remoteClientID := clientMappingConfig.Data[consumer.Status.Client.ID]
+		remoteNamespace := remoteNamespaceByClientID[remoteClientID]
```

Comment:
We should not assume that there is a value in `client.ID` on the consumer's status